zulip/integrations/bridge_with_matrix/matrix_bridge.py

```diff
@@ -212,7 +218,14 @@ def main() -> None:
 
             print("Starting message handler on Zulip client")
             zulip_client.call_on_each_message(zulip_to_matrix(zulip_config, room))
-        except Exception:
+
+        except Bridge_MatrixException as exception:
+            sys.exit("Matrix error: {}".format(exception))
+        except Bridge_ZulipException as exception:
+            sys.exit("Zulip error: {}".format(exception))
+        except zulip.ZulipError as exception:
+            sys.exit("Zulip error: {}".format(exception))
```

Comment:
I think we should have names like `Bridge_ZulipFatalException` if these are going to be fatal, and only use them for things that are structurally fatal, like invalid API key.  Otherwise, I worry this bot will kill itself by accident.

Reply:
I have renamed the internal exceptions to include `Fatal`. Generally, however, I'm not sure which exceptions are *not* fatal, except maybe the very general `except` clause.

From what I can tell, `ZulipError` is not recoverable?

Reply:
I think some transient network errors might come back as `ZulipError`?

Reply:
@timabbott If you mean from the Zulip python API (ie. `__init__.py`), there's:
- `Client.__init__` -> `ZulipError`, `ConfigNotFoundError`, `MissingURLError`
- `Client.do_api_query` -> `ZulipError`
Of those, the latter appears to occur only from a `requests` `ConnectionError` if there has been no server connection (~line 582), which I would take as non-transient?